com.archimatetool.reports/src/com/archimatetool/reports/html/HTMLReportExporter.java

```diff
@@ -325,9 +327,24 @@ private void writeElements(File elementsFolder, ST stFrame, List<EObject> list)
      */
     private void writeElement(File elementFile, ST stFrame, EObject component) throws IOException {
         stFrame.remove("element"); //$NON-NLS-1$
-        //frame.remove("children");
+        stFrame.remove("usedInViews"); //$NON-NLS-1$
+        stFrame.remove("modelRelations"); //$NON-NLS-1$
         stFrame.add("element", component); //$NON-NLS-1$
         
+        if (component instanceof IArchimateElement element) {
+        	stFrame.add("usedInViews", //$NON-NLS-1$
+        			element.getReferencingDiagramObjects().stream()
+        		.map(d -> d.eContainer()).toList()
+    		);
```

Comment:
> Hi, this part returns the `eContainer` of `d`, the diagram object. However, the eContainer is not always the diagram (view), it can be another diagram object. Use this instead:
> 
> ```java
>         	stFrame.add("usedInViews", //$NON-NLS-1$
>         	        DiagramModelUtils.findReferencedDiagramsForArchimateConcept(element));
> ```

Latest commit contains proposed improvement. 

I considered that earlier, having hesitations setting dependencies between different presenters as editor or htmlexporter or jasperreports; i guess you being more familiar with the code base, you know where shared-commons are.

Reply:
I can understand your concern, but this plug-in has a dependency on the editor anyway.